File: 2225-find-players-with-zero-or-one-losses/2225-find-players-with-zero-or-one-losses.cpp
class Solution {
public:
    vector<vector<int>> findWinners(vector<vector<int>>& matches) {
        map<int, int> winner;
        map<int, int> loser;
        for (vector<int>& match : matches) {
            winner[match[0]]++;
            loser[match[1]]++;
        }
        vector<vector<int>> res(2);
        for (auto it : winner) {
            if (loser[it.first] == 0) {
                res[0].push_back(it.first);
            }
        }
        for (auto it : loser) {
            if (it.second == 1) {
                res[1].push_back(it.first);
            }
        }
        return res;
    }
};

File: 2225-find-players-with-zero-or-one-losses/2225-find-players-with-zero-or-one-losses.java
class Solution {
    public List<List<Integer>> findWinners(int[][] matches) {
        List<List<Integer>> result = new ArrayList<>();
        HashSet<Integer> winner = new HashSet<>();
        Map<Integer,Integer> loser = new HashMap<>();

        for (int[] match : matches) {
            loser.put(match[1] , loser.getOrDefault(match[1] , 0) + 1);
        }
        
        for (int[] match : matches) {
            if (!loser.containsKey(match[0])) {
                winner.add(match[0]);
            }
        }

        Iterator<Integer> iterator = winner.iterator();
        List<Integer> list1 = new ArrayList<>();

        while (iterator.hasNext()) {
            list1.add(iterator.next());
        }
        Collections.sort(list1);
        result.add(list1);

        List<Integer> list2 = new ArrayList<>();
        for (Map.Entry<Integer , Integer> map : loser.entrySet()) {
            if (map.getValue() == 1) {
                list2.add(map.getKey());
            }
        }
        Collections.sort(list2);
        result.add(list2);
        return result;
    }
}